src/screens/courses/cardTemplates/FillTheGapCard/index.tsx

```diff
@@ -54,11 +62,31 @@ const FillTheGapCard = ({ isLoading, setIsRightSwipeEnabled }: FillTheGap) => {
 
   useEffect(() => {
     if (!isLoading && !isValidated) {
-      setPropositions(shuffle(card.gapAnswers.map(a => ({ text: a.text, _id: a._id, isSelected: false }))));
-      setSelectedAnswers(goodAnswers.map(() => ''));
+      if (quizzAnswer?.answerList.length) {
+        setPropositions(
+          (quizzAnswer.answerList as StoreAnswerType[])
+            .map(a => ({ _id: a._id, isSelected: a.isSelected, text: a.text }))
+        );
+        setIsValidated(true);
+        setSelectedAnswers(
+          (quizzAnswer.answerList as StoreAnswerType[])
+            .filter(a => a.isSelected)
+            .sort((a, b) => a.position! - b.position!)
```

Comment:
a quoi sert le ! a la fin ?

Reply:
vu qu'on a filtré sur isSelected, position est forcément défini mais TypeScript ne le comprends pas du coup ça permet de lui dire tkt la valeur n'est pas null ou undefined
https://www.geeksforgeeks.org/typescript-non-null-assertion-operator-postfix-type/

Reply:
Ok mais tu as une erreur si tu ne le mets pas ?

Reply:
c'est parce qu'on a le // @ts-nocheck d'activé. Je t'ai fait une proposition où je le retire, dis mois ce que tu en penses

Reply:
@sophiemoustard

Reply:
autre doc qui explique rapidement aussi : https://stackoverflow.com/questions/47704652/what-does-exclamation-point-after-variable-mean-in-typescript

Reply:
Ca me va de l'utiliser ! avec partimoni je pense surtout au debut, le temps qu'on se familiarise avec